### PR TITLE
Feature/design review 2

### PR DESCRIPTION
**Pull request by comountainclimber**
**What current issue(s) from Trello/Github does this address?**
This PR incorporates all of the design review feedback including the new receive functionality **except for the settings UI refactor** It also addresses a number of other items discovered by QA.

- https://github.com/CityOfZion/neon-wallet/issues/1496
- https://github.com/CityOfZion/neon-wallet/issues/1498
- https://github.com/CityOfZion/neon-wallet/issues/1502
- https://github.com/CityOfZion/neon-wallet/issues/1394

It also clears up a bug I found related to abstracts represented as "receive" in tx history (previously it was not showing who it was from just always the users wallet address). Furthermore rendering logic in tx history is simplified

The network switch component is also refactored and simplified so that it can propagate network switch event or serve as a simple select

tabbed-receive

**What problem does this PR solve?**

**How did you solve this problem?**

**How did you make sure your solution works?**

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**

- [ ] Unit tests written?

### CODE REVIEW

File: app/components/Inputs/AssetInput/AssetInput.jsx [hunk 1]
```diff
@@ -1,79 +1,44 @@
...
+  selectedAsset: Object
 }
 
 export default class AssetInput extends React.Component<Props, State> {
-  static defaultProps = {
-    onChange: noop,
-    onFocus: noop,
-    onBlur: noop
-  }
+  parsedSymbols = this.props.symbols.map(symbol => ({
```
Comment (evgenyboxer) [inline app/components/Inputs/AssetInput/AssetInput.jsx]: I think its better if this was a function that would get symbols and check if its an array and then return a mapped result (outside of the class)

File: app/components/Modals/TokenModal/utils.js [hunk 2]
```diff
@@ -8,14 +8,28 @@ export const getNewTokenItem = (networkId: string) => ({
   isUserGenerated: true
 })
 
+const ensureHex = (token: string): boolean => {
+  const hexRegex = /^([0-9A-Fa-f]{2})*$/
```
Comment (evgenyboxer) [inline app/components/Modals/TokenModal/utils.js]: Its better to move the `hexRegex` declaration out of the function (no need to recreate it every time, just reuse it)

Comment (comountainclimber) [reply app/components/Modals/TokenModal/utils.js]: 👍 will definitely move this to a util file (I meant to do that but was moving quickly) thanks

File: app/components/Inputs/AssetInput/AssetInput.jsx (hunk 1 above)
Comment (comountainclimber) [reply app/components/Inputs/AssetInput/AssetInput.jsx]: 👍

**Pull request merged**